examples/imagenet_cnn/functions.py

```diff
@@ -0,0 +1,164 @@
+import tensorflow as tf
+import numpy as np
+import ray
+
+@ray.remote(16 * [np.ndarray], [])
+def updateweights(*weight):
+  feed_dict = dict(zip(placeholders, weight))
+  sess.run(assignment, feed_dict=feed_dict)
+
+@ray.remote([np.ndarray, np.ndarray], 16 * [np.ndarray])
+def computegrad(X, Y):
+  return sess.run([g for (g,v) in compgrads], feed_dict={images:X, y_true:Y, dropout:0.5})
+
+def printaccuracy(X, Y):
+    print sess.run(accuracy, feed_dict={images:X, y_true:Y, dropout:1.0})
+
+def setupvariables(kernelshape, biasshape):
+  kernel = tf.Variable(tf.zeros(kernelshape, dtype=tf.float32))
+  biases = tf.Variable(tf.constant(0.0, shape=biasshape, dtype=tf.float32),
+                       trainable=True, name='biases')
+  kernel_new = tf.placeholder(tf.float32, shape=kernel.get_shape())
+  biases_new = tf.placeholder(tf.float32, shape=biases.get_shape())
+  update_kernel = kernel.assign(kernel_new)
+  update_biases = biases.assign(biases_new)
+  return [kernel, biases], [kernel_new, biases_new], [update_kernel, update_biases]
+
+  
+
+
+images = tf.placeholder(tf.float32, shape=[None, 224, 224, 3])
+y_true = tf.placeholder(tf.float32, shape=[None, 1000])
+parameters = []
+assignment = []
+placeholders = []
+# conv1
+with tf.name_scope('conv1') as scope:
+  newvars = setupvariables([11, 11, 3, 96], [96])
```

Comment:
can you say param, vars, assig = setupvariables(...) and then replace parameters[-x] below by the respective variable? That would be clearer I think.